runtime/lua/vim/treesitter/query.lua

```diff
@@ -8,6 +8,85 @@ Query.__index = Query
 
 local M = {}
 
+-- Some treesitter grammars extend others.
+-- We can use that to import the queries of the base language
+local base_language_map = {
+  cpp = {'c'},
+  typescript = {'javascript'},
+  javascript = {'jsx'},
+  tsx = {'typescript', 'javascript', 'jsx'},
+  ocaml_interface = {'ocaml'},
+}
```

Comment:
Isn't that language map bound to specific parsers? If those parsers are not specified here, maybe this map shouldn't be set or at least be accessible for plugins that want to install parsers or may have alternative parser grammars that don't follow that scheme. So maybe export this variable.

Reply:
Maybe we can expose a function like : 
```
function M.add_language_extension(base_lang, extensions)
```

Where `extension` is used to extend the corresponding key in the base table.

That would allow to remove those defaults here and have them implemented elsewhere (like `nvim-treesitter`).

Reply:
Your comment should be addressed, tell me if this is the API you thought of.